test/AutoDiff/Sema/DerivedConformances/derived_differentiable.swift

```diff
@@ -13,8 +13,8 @@ struct GenericTangentVectorMember<T: Differentiable>: Differentiable,
 // CHECK-AST:   internal var x: T.TangentVector
 // CHECK-AST:   internal static func + (lhs: GenericTangentVectorMember<T>, rhs: GenericTangentVectorMember<T>) -> GenericTangentVectorMember<T>
 // CHECK-AST:   internal static func - (lhs: GenericTangentVectorMember<T>, rhs: GenericTangentVectorMember<T>) -> GenericTangentVectorMember<T>
-// CHECK-AST:   @_implements(Equatable, ==(_:_:)) internal static func __derived_struct_equals(_ a: GenericTangentVectorMember<T>, _ b: GenericTangentVectorMember<T>) -> Bool
```

Comment:
Why did these all move 🤔?

Reply:
Good question. I found [this comment](https://github.com/swiftlang/swift/blob/f66a2b11cb27862f6b4ee1688f17ddcecf357fe0/lib/AST/Decl.cpp#L5995-L5997) and assumed `isSynthesized` must play a role in sorting decls.

Reply:
I think this is why. It looks like members with `isSynthesized` are sorted separate from others, and then merged.

https://github.com/swiftlang/swift/blob/f66a2b11cb27862f6b4ee1688f17ddcecf357fe0/lib/Sema/TypeCheckDecl.cpp#L2940-L2969

Reply:
I assume this is fine, but CC @slavapestov in case the ordering here matters.

Reply:
IIRC the reason Slava added this code to sort implicit members by mangled name was so that the order of members is deterministic, which helps with reproducing build issues. So changing the order is fine so long as the order is still deterministic.

Reply:
Thanks Holly, LGTM then. Thanks @dylansturg!